packages/protocol/contracts/tokenvault/ERC1155Vault.sol

```diff
@@ -122,11 +116,13 @@ contract ERC1155Vault is BaseNFTVault, ERC1155ReceiverUpgradeable {
         // Check context validity
         IBridge.Context memory ctx = checkProcessMessageContext();
 
-        address _to = to == address(0) || to == address(this) ? from : to;
+        // Don't allow sending to disallowed addresses.
+        // Don't send the tokens back to `from` because `from` is on the source chain.
+        if (to == address(0) || to == address(this)) revert VAULT_INVALID_TO();
```

Comment:
Same issue here with using `from` from the source chain on the destination chain. It's safer to let this fail and let the user recall on the source chain.